Maple2.Server.DebugGame/Graphics/Scene/FreeCameraController.cs

```diff
@@ -0,0 +1,342 @@
+
+        Camera.Transform.Position += offset;
+        CameraTarget += offset;
+        UpdateViewMatrix();
+
+        // Unlock camera follow when manually moving
+        if (IsFollowingPlayer && offset.LengthSquared() > 0.01f) {
+            UnlockCameraFollow();
+        }
+    }
+
+    /// <summary>
+    /// Starts following a specific player
+    /// </summary>
+    public void StartFollowingPlayer(long playerId) {
+        IsFollowingPlayer = true;
+        FollowedPlayerId = playerId;
+        HasManuallyStopped = false; // Reset manual stop flag when starting to follow
+        SetDefaultRotation(); // Set default rotation when starting to follow
+        Logger.Information("Started following player {PlayerId}", playerId);
+    }
+
+    /// <summary>
+    /// Stops following the current player
+    /// </summary>
+    public void StopFollowingPlayer() {
+        IsFollowingPlayer = false;
+        FollowedPlayerId = null;
+        HasManuallyStopped = true; // Remember that user manually stopped
+        Logger.Information("Stopped following player");
+    }
+
+    /// <summary>
+    /// Unlocks camera follow when manual movement is detected
+    /// </summary>
+    private void UnlockCameraFollow() {
+        if (IsFollowingPlayer) {
+            IsFollowingPlayer = false;
+            Logger.Information("Camera follow unlocked - manual movement detected");
+        }
+    }
+
+    /// <summary>
+    /// Updates camera to follow a player at the specified position
+    /// </summary>
+    public void UpdatePlayerFollow(Vector3 playerPosition) {
+        if (!IsFollowingPlayer) return;
+
+        SetCameraTarget(playerPosition);
+        SetCameraPosition(playerPosition + CameraFollowOffset);
+        SetDefaultRotation(); // Always use default rotation when following player
+    }
+
+    /// <summary>
+    /// Toggles wireframe rendering mode
+    /// </summary>
+    public void ToggleWireframeMode() {
+        WireframeMode = !WireframeMode;
+        Logger.Information("Wireframe mode: {Mode}", WireframeMode ? "ON" : "OFF");
+    }
+
+    /// <summary>
+    /// Sets camera orientation that works well with MapleStory 2's coordinate system
+    /// </summary>
+    public void SetCameraOrientationForMapData() {
+        // Set camera orientation that works well with the transformed map coordinate system
+        Vector3 offset = new Vector3(0, -5, 5); // Look up from below and in front (inverted from typical)
+        SetCameraPosition(CameraTarget + offset);
+    }
+
+    /// <summary>
+    /// Flips the camera's up vector to fix upside-down view
+    /// </summary>
+    public void FlipCameraUpVector() {
+        if (Camera == null) return;
+
+        // Flip the up axis in the transform
+        Matrix4x4 currentTransform = Camera.Transform.Transformation;
+        Matrix4x4 flipMatrix = Matrix4x4.CreateScale(1, 1, -1);
+        Camera.Transform.Transformation = currentTransform * flipMatrix;
+        UpdateViewMatrix();
+    }
```

Comment:
Also should be in a follow camera controller instead of free camera controller cause its not relevant to free camera controls